pycaret/internal/ts_tests.py

```diff
@@ -0,0 +1,227 @@
+from typing import Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+
+def find_trend(
+    data: np.ndarray,
+    trend_order: int = 1,
+    **kwargs
+):
+    coeffs = np.polyfit(range(len(data)), data, trend_order)
+    slope = coeffs[-2]
+    return slope
+
+
+def trend_and_seasonality_test(
+    ts: Union[pd.DataFrame, pd.Series],
+    seasonal_periods=(3, 5, 7, 12, 14, 30, 31),
+    plots=False,
+    **kwargs
+):
+    """
+
+    Parameters
+    ----------
+    ts : Time series data
+    seasonal_periods : Seasonal periods to test for seasonality
+    plots : When set to True, returns the trend and seasonal plot for the given time series
+
+    Returns
+    -------
+    Test result dict
+
+    """
+    from statsmodels.tsa.seasonal import seasonal_decompose
+
+    # Iterate through all the seasonal periods and best the seasonal period
+    seasonal_values = {i: np.max(seasonal_decompose(ts, period=i).seasonal) for i in seasonal_periods}
+    df = pd.DataFrame(index=seasonal_values.keys(), data=seasonal_values.values(), columns=['values'])
+
+    is_seasonal = np.any(df[['values']] > 1)
+    seasonal_period = df.idxmax()[0] if is_seasonal else None
+
+    decompose_result = seasonal_decompose(ts, period=seasonal_period)
+    t_values = decompose_result.trend.fillna(method='bfill').fillna(method='ffill')
+
+    trend = find_trend(t_values)
+    if plots:
+        f, ax = plt.subplots(nrows=2, ncols=1)
+
+        # trend plot
+        ax[0].plot(ts.index, t_values)
+        ax[0].plot(ts.index, trend * np.arange(1, len(t_values) + 1))
+        ax[0].legend(['Trend from stl', 'Linear trend'])
+        ax[0].set_title('Trend plot', size=12)
+
+        # seasonal plot
+        df.plot.bar(ax=ax[1], rot=0)
+        plt.hlines(1, -1, len(df.index), linestyles='--', colors='black', alpha=0.5)
+        ax[1].legend(['Significance level', 'Seasonal value'])
+        ax[1].set_title('Seasonal plot', size=12)
+
+        f.tight_layout()
+
+    if abs(trend) >= 1:
```

Comment:
Is there any reason to use 0.1 and 1 as the cutoff for the slope to define strong and weak increasnig?